File: javascript/codeeval/169_filename_pattern.js
var fs  = require("fs");

function fp(inp) {
    inp = inp.trim().split(' ')
    r = inp.shift().replace(/\./g, '\\.').replace(/\?/g, '.').replace(/\*/g, '.*')
    r = new RegExp('^(' + r + ')$')
    result = []
    for (i = 0; i < inp.length; i++) {
        if (inp[i].match(r)) {
            result.push(inp[i])
        }
    }
    return result.length > 0 ? result.join(' ') : '-'
}

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        console.log(fp(line))
    }
});

File: python/codeeval/169_filename_pattern.py
import re
import sys

def fp(test):
    inp = test.split(' ')
    r = inp.pop(0)
    r = re.sub('\*', '.*', re.sub('\?', '.', re.sub('\.', '\.', r))) + '$'
    result = []
    for i in inp:
        if re.match(r, i):
            result.append(i)
    return ' '.join(result) if len(result) > 0 else '-'

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        print(fp(test))
